### DISCUSSION TITLE
Add tests to dynamic cross modules example

### DISCUSSION THREAD

**Discussion by asarkar** [com-lihaoyi/mill#3489] [General]
The [dynamic cross modules example](https://github.com/com-lihaoyi/mill/tree/main/example/cross/9-dynamic-cross-modules) doesn't include any tests, so, the example doesn't work when tests are added to any of the modules. Please add some tests to one or more of the modules so that we have a mix of `test` and `run` targets, which is more practical and useful. This will, of course, also necessitate updating the `build.sc` file.

**Comment by asarkar** [comment]
I created a minimal example project to the best of my understanding how dynamic cross module tests can be discovered. I'm attaching it for reference.
[cross.zip](https://github.com/user-attachments/files/16684024/cross.zip)

**Comment by lihaoyi** [comment]
Converted this to a discussion since it's not concrete enough to be an issue yet

**Reply by asarkar** [reply]
@lihaoyi I’ve already attached a sample project that is working code, and contains tests that the current dynamic cross module example doesn’t. It’d appear a practical example is concrete enough. If there’s something more I can do, please let me know.

**Reply by lihaoyi** [reply]
The problem with your build appears to be that your cross modules are not inside a `modules/` folder matching the `object modules` name

**Reply by asarkar** [reply]
Indeed, and I don’t know why that’d be a problem since nothing dictates such a structure. The example uses a `modules` folder that just happens to match the name of the object, so, it’s not immediately apparent that the name of the object is part of the `millSourcePath`, I found that only by trial and error and debugging.

I think this has greater implication when per-module build files (`module.sc`) are present. I understand that work is ongoing in that area, but such a project has two options:
1) Some modules opt-out of common configuration and define their own. For instance, generated code may not be subject to the same checks as written code.
2) Some modules use the common configuration and add to it. For instance, a module may need special dependencies.

I’m very interested in seeing how the above cases will be handled for dynamic modules.
